src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -2029,36 +2029,38 @@ private static object NullRule(CallSite site, object obj)
 
         internal static bool IsValueTypeMutable(Type type)
         {
-            if (type.IsPrimitive || type.IsEnum)
+            // First, check for enums/primitives and compiler-defined attributes.
+            if (type.IsPrimitive
+                || type.IsEnum
+                || type.IsDefined(typeof(System.Runtime.CompilerServices.IsReadOnlyAttribute), inherit: false))
             {
                 return false;
             }
 
-            // If there are any fields, the type is mutable.
-            if (type.GetFields(BindingFlags.Public | BindingFlags.Instance).Any())
+            // If the builtin attribute is not present, check for a custom attribute from by the compiler. If the
+            // library targets netstandard2.0, the compiler can't be sure the attribute will be provided by the runtime,
+            // and defines its own attribute of the same name during compilation.
+            foreach (var attribute in type.GetCustomAttributes(inherit: false))
```

Comment:
`var` -> type

Reply:
🤔 `GetCustomAttributes()` returns only `object[]`. Is it safe to type this var as `Attribute`?

Reply:
If `object` is all it guarantees, then I think `object` makes sense